### PR TITLE
Properly placed pages - Part 1 - Setup architecture

### PR DESCRIPTION
**Pull request by planarvoid**
- create `PagesFragment` containing the search results and view pager with different page filters. 
- add logic that hides view pager and shows search results when the user clicks the magnifying glass
- create layouts and `ViewModels`

pages_redesign_1

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+            linearLayoutManager.onRestoreInstanceState(it)
+        }
+        recyclerView.layoutManager = linearLayoutManager
+
+        swipeToRefreshHelper = WPSwipeToRefreshHelper.buildSwipeToRefreshHelper(pullToRefresh) { viewModel.refresh() }
+        (activity?.application as WordPress).component()?.inject(this)
+
+        val key = arguments?.getString(fragmentKey)
+        viewModel = ViewModelProviders.of(activity!!, viewModelFactory)
+                .get<PageListViewModel>(checkNotNull(key), PageListViewModel::class.java)
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt]: no reason. I should probably use one or the other. `checknotNull` seems more implicit.

File: WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        return inflater.inflate(R.layout.pages_list_fragment, container, false)
+    }
+
+    override fun onDestroyView() {
+        viewModel.detach()
+        super.onDestroyView()
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/pages/PageListFragment.kt]: good idea!

File: WordPress/src/main/java/org/wordpress/android/ui/pages/PagesFragment.kt [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+class PagesPagerAdapter(val context: Context, fm: FragmentManager) : FragmentPagerAdapter(fm) {
+    override fun getCount(): Int = 4
+
+    override fun getItem(position: Int): Fragment {
+        return PageListFragment.newInstance("key$position", Type.getType(position))
+    }
+
+    override fun getPageTitle(position: Int): CharSequence? {
+        return Type.getType(position).text.let { context.getString(it) }
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/pages/PagesFragment.kt]: type is nonnull, it crashes if you try to get a position that doesn't exist. That's why you don't need `?`

File: WordPress/src/main/res/layout/pages_list_fragment.xml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+        <android.support.v7.widget.RecyclerView
+            android:id="@+id/recyclerView"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:scrollbars="vertical"/>
+
+    </org.wordpress.android.util.widgets.CustomSwipeRefreshLayout>
+
+    <org.wordpress.android.widgets.WPTextView
```
[the remark this answers is not in the document]
Comment (planarvoid) [reply WordPress/src/main/res/layout/pages_list_fragment.xml]: good point! what I think I'll do is adding an `empty view` as an item in the list. However it's not part of this PR so for now I'm just removing the empty view.

**Comment by malinajirka** [comment]
Thanks @planarvoid ! LGTM

I've noticed the search query doesn't survive configuration change, but as we discussed on Slack it's out of the scope of this PR.

🚢

**Pull request merged**